runtime/tchannel_server.go

```diff
@@ -177,6 +177,9 @@ func (s *TChannelRouter) handle(ctx context.Context, handler handler, service st
 		if er := reader.Close(); er != nil {
 			return errors.Wrapf(er, "could not close arg3reader for inbound call: %s::%s", service, method)
 		}
+		s.logger.Warn("Unexpected tchannel system error",
+			zap.String("error", err.Error()),
+		)
 		return call.Response().SendSystemError(err)
```

Comment:
we do not want to send the `err` out. We want to send a generic "Server Error" message.